### PR TITLE
block-identifier util

### PR DESCRIPTION
**Pull request by pipermerriam**
### What was wrong?

There was repeated and generally not great code to determine which RPC method to use for some of the `web3.eth` APIs

### How was it fixed?

Abstracted it into a more robust generic helper function.

#### Cute Animal Picture

851159422_562cf6467a_z

### CODE REVIEW

**Comment by pipermerriam** [comment]
This needs tests written against it.

File: web3/utils/blocks.py [hunk 1]
```diff
@@ -1,10 +1,44 @@
...
+def is_hex_encoded_block_number(value):
+    if not is_string(value):
+        return False
+    elif is_hex_encoded_block_hash(value):
+        return False
+    try:
+        value_as_int = int(value, 16)
+    except ValueError:
+        return False
+    return 0 <= value_as_int < 2**256 - 1
```
Comment (carver) [inline web3/utils/blocks.py]: off by one, it looks like: `0 <= value_as_int < 2**256`

Comment (pipermerriam) [reply web3/utils/blocks.py]: 👍

**Comment by pipermerriam** [comment]
Tests added.

**Pull request merged**